enzyme/Enzyme/GradientUtils.h

```diff
@@ -79,6 +79,13 @@ extern llvm::StringMap<std::function<llvm::Value *(
     shadowHandlers;
 
 class DiffeGradientUtils;
+
+/// This is the main entry point to register a custom derivative for language frontends.
+/// It is more general, and therefore recommended over trying to register 
```

Comment:
I wouldn't go so far as to recommending it quite yet, it is a fair bit more complex after all